File: BinaryTreeLevelOrderTraversalII/fourth.cpp
class Solution {
public:
    int height(TreeNode *root){
        if(root==NULL)
            return 0;
        return max(height(root->left),height(root->right))+1;
    }
    void fill_level(vector<int>* v,TreeNode* root,int level){
        if(root==NULL)
            return;
        if(level==1)
        {
            (*v).push_back(root->val);
            return ;
        }
        level=level-1;
        if(root->left!=NULL)
            fill_level(v,root->left,level);
        if(root->right!=NULL)
            fill_level(v,root->right,level);
        return ;
    }
    vector<vector<int>> levelOrderBottom(TreeNode* root) {
        vector<vector<int>> level_order;
        for(int i=height(root);i>0;--i)
        {
            vector<int> v;
            fill_level(&v,root,i);
            level_order.push_back(v);
        }
        return level_order;
    }
}; 

File: BinaryTreeLevelOrderTraversalII/fourth.java
public class Solution {
    public ArrayList<ArrayList<Integer>> levelOrderBottom(TreeNode root) {
        ArrayList<ArrayList<Integer>> list = new ArrayList<ArrayList<Integer>>();
        ArrayList<ArrayList<Integer>> list2 = new ArrayList<ArrayList<Integer>>();
        if(root==null){
           return list; 
        }
        // Stack<TreeNode> stack = new Stack<TreeNode>();
        // stack.push(root);

        Queue<TreeNode> queue = new LinkedList<TreeNode>();

        queue.offer(root);

        while(!queue.isEmpty()){
            ArrayList<Integer> level = new ArrayList<Integer>();
            int size = queue.size();
            for(int i=0; i<size; i++){
                TreeNode head = queue.poll();
                level.add(head.val);
                if(head.left!=null){
                    queue.offer(head.left);
                }
                if(head.right!=null){
                    queue.offer(head.right);
                }
            }

            list2.add(level);
        }
        for(int i=list2.size()-1; i>=0; i--){
            list.add(list2.get(i));
        }

        return list;

    }
}
